builtin/providers/aws/provider.go

```diff
@@ -12,8 +12,12 @@ import (
 
 	"github.com/aws/aws-sdk-go/aws/credentials"
 	"github.com/aws/aws-sdk-go/aws/credentials/ec2rolecreds"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
 )
 
+var defaultClient = ec2metadata.New(session.New())
```

Comment:
Is there any reason for this to be defined here rather than as part of the allocation of `ec2rolecreds.EC2RoleProvider` on line 49?